src/fns.js

```diff
@@ -0,0 +1,52 @@
+export const shallowMerge = (...objects) => {
+  return objects.reduce((x, y) => {
+    return Object.keys(y)
+      .reduce((o, key) => {
+        o[key] = y[key];
+
+        return o;
+      }, x);
+  }, {});
+};
+
+export const flatten = lists => lists.reduce((x, y) => x.concat(y), []);
+
+export const range = (count, start = 0) => Array(count).fill().map((_, i) => i + start);
+
+// P10Y10M10DT10H10M10.1S
+export const parseDuration = str => {
+  const durationRegex = /^([-])?P(([\d.]*)Y)?(([\d.]*)M)?(([\d.]*)D)?T?(([\d.]*)H)?(([\d.]*)M)?(([\d.]*)S)?/;
+
+  const SECONDS_IN_YEAR = 365 * 24 * 60 * 60;
+  const SECONDS_IN_MONTH = 30 * 24 * 60 * 60;
+  const SECONDS_IN_DAY = 24 * 60 * 60;
+  const SECONDS_IN_HOUR = 60 * 60;
+  const SECONDS_IN_MIN = 60;
+  const match = durationRegex.exec(str);
```

Comment:
Should we do anything if the string that's passed in is malformed? Or is assuming the value is 0 enough?